tern/analyze/common.py

```diff
@@ -452,3 +453,61 @@ def get_os_style(image_layer, binary):
                     package_manager=binary,
                     package_format=image_layer.pkg_format,
                     os_list=image_layer.os_guess), 'info'))
+
+
+def check_git_src(dockerfile_path, src_path):
+    '''Given the src_path and the dockerfile path, return the git
+    repository name and sha information in the format of string.
+    Currently we only consider the following situation:
+    - target_git_project
+        - dir1
+        - dir2/dockerfile
+    So we only use dockerfile_path to find the git repo info.'''
+    # get the path of the folder containing the dockerfile
+    dockerfile_folder_path = os.path.dirname(dockerfile_path)
+    # locate the top level directory
+    path_to_toplevel = get_git_toplevel(dockerfile_folder_path)
```

Comment:
What happens if we just do `get_git_toplevel(dockerfile_path)`? Shouldn't we get the same thing? If so, this would eliminate having to create a `dockerfile_folder_path` variable that is only used once.

Reply:
I set the `cwd(current wording diretory)` to be `dockerfile_folder_path` in `subprocess.check_output` when looking for toplevel dir. `dockerfile_path` points to a file, not a diretory which cannot be set as `cwd`.